File: src/main/java/com/example/drycleaner/model/Image.java
package com.example.drycleaner.model;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "images")
@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Image {
    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(name = "image_id", columnDefinition = "BINARY(16)", nullable = false, updatable = false)
    UUID imageId;

    @Column(name = "created_date")
    @CreationTimestamp
    LocalDateTime createdDate;

    @Column(name = "updated_date")
    @UpdateTimestamp
    LocalDateTime updatedDate;

    @Column(name = "image_url")
    String imageUrl;

    @ManyToOne
    @JoinColumn(referencedColumnName = "dry_cleaner_id")
    DryCleaner dryCleaner;
}

File: src/main/java/com/reserveall/drycleaner/model/Image.kt
package com.reserveall.drycleaner.model

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.UpdateTimestamp
import java.time.LocalDateTime

@Entity
@Table(name = "images")
data class Image(
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(name = "id")
    val id: String?,

    @Column(name = "created_date")
    @CreationTimestamp
    val createdDate: LocalDateTime?,

    @Column(name = "updated_date")
    @UpdateTimestamp
    val updatedDate: LocalDateTime?,

    @Column(name = "image_url")
    val imageUrl: String?,

    @ManyToOne
    @JoinColumn(name = "dry_cleaner_id", referencedColumnName = "id")
    val dryCleaner: DryCleaner?
)
